torch/onnx/utils.py

```diff
@@ -231,7 +235,16 @@ def _model_to_graph(model, args, f, verbose=False, training=False,
             raise RuntimeError('\'forward\' method must be a script method')
     else:
         graph, torch_out = _trace_and_get_graph_from_model(model, args, training)
-        params = list(_unique_state_dict(model).values())
+        state_dict = _unique_state_dict(model)
+        params = list(state_dict.values())
+        if retain_param_name:
+            graph_inputs = list(graph.inputs())
+            user_input_num = len(graph_inputs) - len(state_dict)
+            param_names = list(state_dict.keys())
+            for i in range(len(graph_inputs)):
+                if i >= user_input_num:
+                    graph_inputs[i].setUniqueName(
```

Comment:
what happens if there's a duplicated name?

Reply:
https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/ir.cpp#L670, it will rename the old owner of this name. We should be good. :-)